### PR TITLE
github-action: undeploy my kibana

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

GitHub composite action to centralise the logic for undeploying Kibana PRs

## Why is it important?

Keep it simple

### CODE REVIEW

File: .github/actions/undeploy-my-kibana/action.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        url: ${{ inputs.vault-url }}
+        roleId: ${{ inputs.vault-role-id }}
+        secretId: ${{ inputs.vault-secret-id }}
+
+    - name: Is an Elastician comment?
+      id: is_elastic_member
+      uses: elastic/apm-pipeline-library/.github/actions/is-member-elastic-org@current
+      with:
+        username: ${{ inputs.user }}
+        token: ${{ env.GITHUB_TOKEN }}
```
Comment (kuisathaverat) [inline .github/actions/undeploy-my-kibana/action.yml:37]: it should be parameter

File: .github/actions/undeploy-my-kibana/action.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        gh issue \
+          create \
+          --label 'destroy-custom-kibana-serverless' \
+          --title "[Undeploy Kibana] for user ${{ inputs.user }} with PR kibana@pr-${{ inputs.pull-request }}" \
+          --body-file .body-content \
+          --repo elastic/observability-test-environments | tee .issue
+        echo "ISSUE=$(cat .issue)" >> $GITHUB_ENV
+        echo "issue=$ISSUE" >> "$GITHUB_OUTPUT"
+      env:
+        GH_TOKEN: ${{ env.GITHUB_TOKEN }}
```
Comment (kuisathaverat) [inline .github/actions/undeploy-my-kibana/action.yml:65]: it should be parameter

File: .github/actions/undeploy-my-kibana/action.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        EOT
+      shell: bash
+
+    - name: Create github issue for the undeploy-my-kibana
+      if: contains(steps.is_elastic_member.outputs.result, 'true')
+      run: |
+        gh issue \
+          create \
+          --label 'destroy-custom-kibana-serverless' \
+          --title "[Undeploy Kibana] for user ${{ inputs.user }} with PR kibana@pr-${{ inputs.pull-request }}" \
```
Comment (kuisathaverat) [inline .github/actions/undeploy-my-kibana/action.yml:59]: Possible injection

File: .github/actions/undeploy-my-kibana/action.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          --body-file .body-content \
+          --repo elastic/observability-test-environments | tee .issue
+        echo "ISSUE=$(cat .issue)" >> $GITHUB_ENV
```
Comment (kuisathaverat) [inline .github/actions/undeploy-my-kibana/action.yml:62]: Where is `.issue` created?

File: .github/actions/undeploy-my-kibana/action.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        username: ${{ inputs.user }}
+        token: ${{ env.GITHUB_TOKEN }}
+
+    - name: Create github issue body
+      if: contains(steps.is_elastic_member.outputs.result, 'true')
+      run: |-
+        cat <<EOT >> .body-content
+        ### Kibana branch
+
+        ${{ inputs.pull-request }}
```
Comment (kuisathaverat) [inline .github/actions/undeploy-my-kibana/action.yml:45]: Possible injection

File: .github/actions/undeploy-my-kibana/action.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+        ### Further details
+
+        Caused by @${{ inputs.user }} in https://github.com/elastic/kibana/pull/${{ inputs.pull-request }}
```
Comment (kuisathaverat) [inline .github/actions/undeploy-my-kibana/action.yml:49]: possible injection

File: .github/actions/undeploy-my-kibana/action.yml (hunk 2 above)
Comment (v1v) [reply .github/actions/undeploy-my-kibana/action.yml:65]: I could change the signature and use the new approach, let me do it

File: .github/actions/undeploy-my-kibana/action.yml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    required: true
+  pull-request:
+    description: 'The GitHub Pull Request ID'
+    default: ${{ github.event.pull_request.number }}
+  repository:
+    description: 'The GitHub repository'
+    default: ${{ github.repository }}
+  user:
+    description: 'The GitHub PR onwer'
+    default: ${{ github.event.pull_request.head.repo.owner.login }}
```
[the remark this answers is not in the document]
Comment (v1v) [reply .github/actions/undeploy-my-kibana/action.yml:22]: Wanted to support other cases, such as running the action if no a PR but using a PR values

File: .github/actions/undeploy-my-kibana/action.yml (hunk 3 above)
Comment (v1v) [reply .github/actions/undeploy-my-kibana/action.yml:59]: I'll use envs, good catch

File: .github/actions/undeploy-my-kibana/action.yml (hunk 6 above)
Comment (v1v) [reply .github/actions/undeploy-my-kibana/action.yml]: See https://github.com/elastic/apm-pipeline-library/pull/2595/commits/f69afa9a7c11b62197ecd4651cf2512b26c1a273

File: .github/actions/undeploy-my-kibana/action.yml (hunk 5 above)
Comment (v1v) [reply .github/actions/undeploy-my-kibana/action.yml]: See https://github.com/elastic/apm-pipeline-library/pull/2595/commits/72f9beafc74a61255d058a63a808a34c7edd42b2

File: .github/actions/undeploy-my-kibana/action.yml (hunk 3 above)
Comment (v1v) [reply .github/actions/undeploy-my-kibana/action.yml]: See https://github.com/elastic/apm-pipeline-library/pull/2595/commits/f69afa9a7c11b62197ecd4651cf2512b26c1a273

File: .github/actions/undeploy-my-kibana/action.yml (hunk 2 above)
Comment (v1v) [reply .github/actions/undeploy-my-kibana/action.yml]: See https://github.com/elastic/apm-pipeline-library/pull/2595/commits/f69afa9a7c11b62197ecd4651cf2512b26c1a273

File: .github/actions/undeploy-my-kibana/action.yml (hunk 1 above)
Comment (v1v) [reply .github/actions/undeploy-my-kibana/action.yml]: See https://github.com/elastic/apm-pipeline-library/pull/2595/commits/f69afa9a7c11b62197ecd4651cf2512b26c1a273

**Review by reakaleek — APPROVED** [review]

File: .github/actions/undeploy-my-kibana/action.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (kuisathaverat) [reply .github/actions/undeploy-my-kibana/action.yml:13]: it opens the possibility to skip the pass that checks the PR owner

[the remark this answers is not in the document]
Comment (v1v) [reply .github/actions/undeploy-my-kibana/action.yml:13]: https://github.com/elastic/apm-pipeline-library/pull/2600 is now the way, implicitly detect the PR author

**Pull request merged**